src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/ElasticSearchLogStorageFactory.java

```diff
@@ -87,15 +89,123 @@ public TaskListener nodeListener(FlowNode node) throws IOException, InterruptedE
     @Override
     public AnnotatedLargeText<Executable> overallLog(Executable build, boolean complete)
     {
-      ByteBuffer buf = new ByteBuffer();
-      return new AnnotatedLargeText<>(buf, StandardCharsets.UTF_8, true, build);
+      if(!config.isReadLogsFromElasticsearch()) {
+        AnnotatedLargeText<Executable> logFromFile = tryReadLogFile(build, complete);
+        if(logFromFile != null && logFromFile.length() > 0) {
```

Comment:
we have a lot of times almost identical coding if statements.
Might be worth looking into simplifying this.

Reply:
I tried to simplify the logic as much as possible now. It would be nice if you could confirm I did not miss something.